In [1]:
import pandas as pd
import numpy as np
import scipy
from scipy import stats

In [2]:
ls

 Volume in drive C has no label.
 Volume Serial Number is 703C-6D5B

 Directory of C:\Users\lenovo1

03/05/2022  01:30 PM    <DIR>          .
03/05/2022  01:30 PM    <DIR>          ..
03/05/2022  01:28 PM    <DIR>          .ipynb_checkpoints
02/27/2022  08:54 PM    <DIR>          .ipython
02/26/2022  04:07 PM    <DIR>          .jupyter
02/26/2022  02:10 PM    <DIR>          Anaconda3
03/05/2022  01:07 PM            33,099 Assignment 4 (toyoto corolla) (1).ipynb
03/05/2022  01:07 PM           124,835 Assignment clusters (Crimes data) (2).ipynb
03/05/2022  01:07 PM           314,540 assignment company_datadecision tree (1).ipynb
03/05/2022  01:07 PM            77,349 assignment company_datadecision tree.ipynb
03/05/2022  01:30 PM               833 Assignment Hypothesis testing.ipynb
03/05/2022  01:14 PM            25,021 assignment NN forest fire.ipynb
02/28/2022  10:10 AM    <DIR>          Assignments
02/26/2022  04:12 PM            17,023 Company_Data.csv
02/27/2022  02:57 AM    <DIR> 

# Cutlet assignment

In [3]:
cutlet=pd.read_csv("Cutlets.csv")

In [4]:
cutlet.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [9]:
a=pd.Series(cutlet["Unit A"])
print(a)

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64


In [11]:
b=pd.Series(cutlet["Unit B"])
print(b)

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64


In [12]:
stats.ttest_ind(a,b)

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

#Hence pvalue is greater than 0.05 so accept null hypothesis

# BuyerRatio Assignment

In [15]:
BR=pd.read_csv('BuyerRatio.csv')

In [16]:
BR.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

# Two Tail Test

In [3]:
male = [50,142,131,70]
female=[435,1523,1356,750]

In [4]:
stats.ttest_ind(male,female)

Ttest_indResult(statistic=-3.5834877367684856, pvalue=0.011594334372357056)

In [12]:
import scipy.stats as stats
chistats=stats.chi2_contingency([male,female])
print(chistats)
print('chi =%f p-value=%f' % (chistats[0], chistats[1]))


(1.595945538661058, 0.6603094907091882, 3, array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))
chi =1.595946 p-value=0.660309


# chi>0.05 and so the p-value
so we accept null hypothesis

# Costomer assignment

In [14]:
import pandas as pd
co=pd.read_csv('Costomer+OrderForm.csv')

In [15]:
co.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [16]:
co.shape

(300, 4)

In [20]:
co['Phillippines'].value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [21]:
co['Indonesia'].value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [22]:
co['Malta'].value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [23]:
co['Malta'].value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [24]:
stats.chi2_contingency([[271,267,269,280],[29,33,31,20]])

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

p-value(0.2771)>0.05
so we accept null hypothesis

# Labtat_Assignment

In [26]:
lab=pd.read_csv("LabTAT.csv")
print(lab)

     Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]


In [27]:
lab.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Laboratory 1  120 non-null    float64
 1   Laboratory 2  120 non-null    float64
 2   Laboratory 3  120 non-null    float64
 3   Laboratory 4  120 non-null    float64
dtypes: float64(4)
memory usage: 3.8 KB


In [29]:
lab.describe()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
count    120.000000    120.000000    120.000000     120.00000
mean     178.361583    178.902917    199.913250     163.68275
std       13.173594     14.957114     16.539033      15.08508
min      138.300000    140.550000    159.690000     124.06000
25%      170.335000    168.025000    188.232500     154.05000
50%      178.530000    178.870000    199.805000     164.42500
75%      186.535000    189.112500    211.332500     172.88250
max      216.390000    217.860000    238.700000     205.18000

In [30]:
lb1=pd.DataFrame(lab["Laboratory 1"])
lb2=pd.DataFrame(lab["Laboratory 2"]) 
lb3=pd.DataFrame(lab["Laboratory 3"])
lb4=pd.DataFrame(lab['Laboratory 4']) 

In [31]:
import scipy.stats as stats
stats.f_oneway(lb1,lb2,lb3,lb4)

F_onewayResult(statistic=array([118.70421654]), pvalue=array([2.11567089e-57]))

Hence: p-value>0.05 so we accept null hypothesis